app/views/courses/revisions.json.jbuilder

```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 json.course do
+  course_specific_revision_ids = @course.tracked_revisions.pluck(:id)
```

Comment:
This could be a significant performance problem for a course with huge numbers of revisions. Is there a way to figure this out as part of the revisions query on the following line, putting that work onto the database instead of Ruby?

Reply:
I'll try to find a way and then reply back!

Reply:
@ragesoss 
I tried to experiment what could be done regarding this, so I thought of using the Rails - and & operator for the intersection and difference of the two queries(all revisions and tracked_revisions), but that would make the queries complex and I also couldn't make sense of how they could possibly help us in improving in the performance problem, so what I now did was send all the course_specific_revision_ids as data in the json sent and the filter out everything on the client side itself, and also limited the number of tracked_revisions (which will be atmost the variable limit, because atmost every revision will be a tracked revision).